### PR TITLE
feat: Add webhooks_base_url to det deploy

### PR DESCRIPTION
**Pull request by julian-determined-ai**
## Description

feat: Add webhooks_base_url to det deploy

This PR exists because I wanted to be able to set the `webhooks_base_url` in latest master. This implements the ability for `det deploy` to set the appropriate configuration option. 

If someone could tell me where exactly I should be placing 

`webhooks_base_url=http://latest-master.determined.ai:8080` for our latest master deploy job that would be super helpful. 

## Test Plan

TBD

## Checklist
- [ ] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
~- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.~
~- [ ] Licenses should be included for new code which was copied and/or modified from any external code.~

### CODE REVIEW

File: harness/determined/deploy/gcp/constants.py [hunk 1]
```diff
@@ -21,3 +21,4 @@ class defaults:
     REGION = "us-west1"
     SCHEDULER_TYPE = "fair_share"
     PREEMPTION_ENABLED = False
+    WEBHOOKS__BASE_URL = ""
```
Comment (stoksc) [inline harness/determined/deploy/gcp/constants.py:24]:
```suggestion
    WEBHOOKS_BASE_URL = ""
```

File: harness/determined/deploy/aws/templates/fsx.yaml [hunk 2]
```diff
@@ -195,6 +195,11 @@ Parameters:
     Type: String
     Description: preexisting FSx ID to mount
     Default: ""
+  
+  WebhooksBaseUrl:
```
Comment (stoksc) [inline harness/determined/deploy/aws/templates/fsx.yaml:199]: i imagine there has to be a way to autopopulate this field for the average deployment (that just uses the public dns record assigned by aws, and doesn't also have a record for, e.g. `latest-master.determined.ai`

**Review by stoksc — APPROVED** [review]
lgtm, though may want to get an infra-eng take

**Comment by stoksc** [comment]
also cc @dzhu of the fact this exists, since he has worked on some similar stuff, he may have an opinion about it? tldr, we added `webhooks.base_url` as a config so we know how to redirect back to ourselves from webhooks. kinda just like the saml/oidc 'where is the master for me to redirect back to' configs. didn't seem worth generalizing.

File: harness/determined/deploy/aws/templates/fsx.yaml (hunk 2 above)
Comment (julian-determined-ai) [reply harness/determined/deploy/aws/templates/fsx.yaml:199]: @stoksc 

If what you said above cannot be implemented, is this the file where I need to define the default URL to `http://latest-master.determined.ai:8080/`?

Comment (julian-determined-ai) [reply harness/determined/deploy/aws/templates/fsx.yaml:199]: Oh yea that makes sense. Thanks!

**Review by dannysauer — APPROVED** [review]
I don't see any huge problem with this, but bear in mind that I'm on vacation, so I don't see a problem with much. ;)

I do think @stoksc 's note about figuring out the hostname is a very good aspiration. That might be a useful ticket for infra to help with figuring out.

**Pull request merged**